utils/defined_error.h

```diff
@@ -24,6 +24,14 @@ typedef enum {
   RET_HTTP_CERT,    /**< failed on x509 cert parse */
   RET_HTTP_CONNECT, /**< failed on HTTP initial connection */
   RET_HTTP_SSL,     /**< failed on setting ssl config */
+  RET_DEVICE_REG,   /**< register device error */
```

Comment:
Don't use `RET_DEVICE_REG`. Instead, you can use `RET_DEVICE_INIT`. Similarly, `RET_DEVICE_FINI`.
@HowJMay, prefix `RET_` might be a bit misleading since we mention errors/warning here. How about changing it to `ERR_`?

Reply:
The prefix misleading problem has been discussed last week. This issue has been noted on todo list inside hackmd.

Reply:
`REG` is usually used for registers. Don't confuse.

Reply:
?

Reply:
If you are going to ask someone for reviewing, you must resolve all concerns during past discussions.